docs/process/ci_build.md

```diff
@@ -0,0 +1,89 @@
+# Continuous Integration and Builds
+
+## Overview
+
+Currently, we use Github Actions to run our continuous integration & builds.
+
+Basic concepts:
+
+* A step is the smallest unit of work in Github Actions. It can be as simple as running a shell command.
+* A step is versatile because it can also invoke external [community-contributed functions](https://github.com/marketplace?type=actions) ("actions"). [Actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/introduction-to-github-actions) can be used for simple operations like checking out the latest code in the repository, or slightly more complicated tasks like running a benchmarking tool and formatting the results. Actions are usually kept small in scope. They are implemented as repositories on Github whose contents implement interfaces defined by Github Actions.
+* A job runs multiple steps **serially**.
+* A job groups together related steps that achieve a larger task (ex: run unit tests)
+* Each job instance is run in a completely new VM instance ("runner").
+* A workflow runs multiple jobs **in parallel**.
+* Workflows are run by Github Actions any time certain trigger events occur. Each workflow configures what trigger events conditions must be met for it to to be invoked (ex: only for Pull Requests; only on merges to the main branch)
+* Jobs within a workflow can be made to run in a particular order by specifying job dependencies. This creates the directed acyclic graph (DAG) relationship of job execution.
+
+## Getting Started
+
+* This [intro page](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/introduction-to-github-actions) shows the following
+  * A Github Action workflow is written in YAML syntax, and must be placed in `.github/workflows`.
+  * A separate tab within the Github UI for a repository shows the history of all Actions runs for the repository.
+  * In the UI, for each workflow run instance, you can see each job of the workflow, and each step within each job.  For a job,you can see the console output (stdout, stderr) for each step with linkable line numbers and a search bar for searching the logs.
```

Comment:
space after ',' at '... for a job,you can ...'

Reply:
Done.